PokemonFactory.sol

```diff
@@ -4,32 +4,112 @@ pragma solidity >=0.7.0 <0.9.0;
 
 contract PokemonFactory {
 
-  struct Pokemon {
-    uint id;
-    string name;
-  }
+    struct Ability {
+        string name;
+        string description;
+    }
+
+    struct Type {
+        uint id;
+        string name;
+    }
+
+    struct Pokemon {
+        uint id;
+        string name;
+        string imageUri;
+        Ability[] abilities;
+        Type[] types;
+        Type[] weaknesses;
+    }
 
     Pokemon[] private pokemons;
 
     mapping (uint => address) public pokemonToOwner;
     mapping (address => uint) ownerPokemonCount;
 
-     function createPokemon (string memory _name, uint _id) public {
-        pokemons.push(Pokemon(_id, _name));
-        pokemonToOwner[_id] = msg.sender;
-        ownerPokemonCount[msg.sender]++;
+    mapping (uint => string) public pokemonTypes;
+    uint public amountOfTypes; 
+
+    event NewPokemon(uint id, string name, string imageUri, Ability[] abilities, Type[] types, Type[] weaknesses);
+
+    constructor(string[] memory _types) {
+        pokemons.push();
+        amountOfTypes = _types.length;
+
+        for(uint i = 0; i < _types.length; i++) {
+            pokemonTypes[i] = _types[i];
+        }
```

Comment:
Porque esto? El Smart contract únicamente se inicializará una vez sola vez, porque hacer un for?

Reply:
Este lo hago para inicializar el mapping de tipos, al constructor le paso un array de string con el nombre de los tipos, aca sino como podria inicializarlo?

Reply:
Y si creamos un método que el usuario agregue tipos? Es opcional claro está, pero me parece que es más usable esta forma, que te parece?